src/components/20-molecules/file-upload/index.js

```diff
@@ -251,78 +296,153 @@ class AXAFileUpload extends LitElement {
         : '';
   }
 
-  validateFiles(compressedImages, notImagesFiles) {
-    const { maxSizeOfSingleFileKB, maxNumberOfFiles } = this;
+  validateFiles(compressedImages, notImagesFiles, droppedFilesWithID) {
+    const { maxSizeOfSingleFileKB } = this;
     const maxSizeOfSingleFileInBytes = getBytesFromKilobyte(
       maxSizeOfSingleFileKB
     );
 
-    const faultyFiles = [];
-    const finalFiles = [];
-    const files = [...notImagesFiles].concat(compressedImages);
+    const faultyCompressedFiles = [];
+    const validCompressedFiles = [];
+    const faultyOriginalFiles = [];
+    const validOriginalFiles = [];
+
+    const newOriginalFiles = droppedFilesWithID;
+    const newCompressedFiles = [...notImagesFiles].concat(compressedImages);
 
-    for (let i = 0, n = files.length; i < n; i++) {
-      const file = files[i];
+    for (let i = 0, n = newOriginalFiles.length; i < n; i++) {
+      const file = newOriginalFiles[i];
       const fileSize = file.size;
       if (fileSize > maxSizeOfSingleFileInBytes) {
-        faultyFiles.push(file);
+        // add original file to array
+        faultyOriginalFiles.push(file);
+
+        // find the compromised file with same id and add it to array
+        for (let j = 0; newCompressedFiles.length > j; j++) {
+          if (newCompressedFiles[j].id === file.id) {
+            faultyCompressedFiles.push(newCompressedFiles[j]);
+          }
+        }
       } else {
-        finalFiles.push(file);
+        // add original file to array
+        validOriginalFiles.push(file);
+
+        // find the compromised file with same id and add it to array
```

Comment:
Code duplication at line 320. Create a function for that. You could take this comment to name this function. (PS: you can perfectly unit-test functions!)

Reply:
done